File: src/MrHotel/Database/Entities/Reservations/ReservationInfo.cs
namespace MrHotel.Database.Entities.Reservations;

using MrHotel.Database.Entities.Guests;
using MrHotel.Database.Entities.Rooms;

public partial class ReservationInfo
{
    public Guid Id { get; init; }

    public GuestInfo Guest { get; set; } = null!;

    public RoomInfo Room { get; set; } = null!;

    public required DateTimeOffset CheckInDate { get; set; }

    public required DateTimeOffset CheckOutDate { get; set; }

    public bool CheckOutDone { get; set; }

    public float Price { get; set; }
}

File: src/MrHotel/WebApp/src/app/services/room-manager/data/reservation-info.ts
import { Guid } from "@customTypes/.";
import { RoomInfo } from ".";
import { GuestInfo } from "@services/guest-manager/data";

export type ReservationInfo = {
  id: Guid;
  guest: GuestInfo;
  room: RoomInfo;
  checkInDate: Date;
  checkOutDate: Date;
  price: number;
};
